test/runtests.jl

```diff
@@ -346,6 +346,29 @@ end
     test_broadcast(a_disk1)
 end
 
+@testset "zip" begin
+    da = _DiskArray(rand(10, 9, 2); chunksize=(5, 3, 2))
+    a = collect(da)
+    zd = zip(da, da)
+    za = zip(a, a)
+    @test collect(zd) == collect(za)
+    @test all(zd .== za)
+    a = collect(da)
+    @test zip(a, da, a) isa DiskArrays.DiskZip
+    @test zip(da, da, a) isa DiskArrays.DiskZip
+    @test zip(da, da, da) isa DiskArrays.DiskZip
+    @test zip(a, da, da) isa DiskArrays.DiskZip
```

Comment:
It would be really good to check how many getindex/setindex calls are done into the underlying arrays. Could we try to diagnose the `getindex_count(da)` to understand what is happening?

Reply:
Will have a look

Reply:
I added `getindex_count` checks to the tests, its only `6` per array - so only chunked reads